persistence/src/main/java/edu/softserve/zoo/persistence/provider/impl/HibernatePersistenceProvider.java

```diff
@@ -133,15 +133,15 @@ public boolean delete(Long id, Class type) {
      */
     @Override
     @Transactional(propagation = Propagation.MANDATORY, readOnly = true)
-    public List<T> find(Specification<T> specification) {
+    public <K> List<K> find(Specification<T> specification) {
```

Comment:
K doesn't relate to T if K is set manually, otherwise method returns List<T>. K is used for custom return of specification, e.g. Long or Map, etc. If given generic type isn't connected to Specification return type, there will be Cast exception